### ISSUE TITLE
[Accessibility]A11y_Java Tooling for IntelliJ/Eclipse - New Features_ Spark Local Console – Locally run _Keyboard: When focus is on “Close” button, for 2 tabs focus is not visible before going to “Auto fix” button.

### ISSUE THREAD

**Issue by v-lasiva** [microsoft/azure-tools-for-java#3606]
[Check out Accessibility Insights!](url) - Identify accessibility bugs before check-in and make bug fixing faster and easier.

GitHubTags: #A11yTCS; #A11yMAS; #DesktopApp; #Win10; #A11y_JavaToolingforIntelliJ/Eclipse–New Feature; #MAS2.4.3; #Keyboard; #C+E-H; #JavatoolingforIntelliJ/Eclipse;

**Environment Details:**
URL: https://www.jetbrains.com/idea/download/#section=windows
Version: 2019.2.2
Build: 192.6603.28
MAS Violated: MAS 2.4.3

**Repro Steps:**
1.Download IntelliJ application from the URL:  https://www.jetbrains.com/idea/download/#section=windows .
2.Launch IntelliJ IDEA in Light theme.
3.Tab till Create New Project button and press enter
4.Tab till Azure Spark/HDInsight-->Spark Project Scala
5.Tab till Build Tools-->Maven
6.Tab till Next button and press enter
7.Tab till finish button and press enter
8. Navigate to Run --> Edit Configurations-->Apache Spark on HDInsight > [Spark on HDInsight] myApp
9.Tab till Locally run tab and press enter
10.Fill all the required details properly and tab till Ok button and press enter.
11.Tab till Tools->Spark console ->Run Spark Console(scala) and press enter.
12.Verify whether focus order is following according to the controls present in the page or not.

**Actual Result:**
When focus is on “Close” button, for 2 tabs focus is not visible before going to “Auto fix” button.

**Expected Result:**
When focus is on “Close” button, If we tab focus should immediately point to the next interactive control present in the page

**User Impact:**
Keyboard users will face difficulty in knowing where the focus is pointing if focus is not visible for 2 tabs after “Close” button.

MAS Reference:
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0de7fbe1-ad7e-48e5-bcbb-8d986691e2b9}

[Bug_Focus order issue after close button before going to auto fix button in Run local spark console.pptx](https://github.com/microsoft/azure-tools-for-java/files/3645449/Bug_Focus.order.issue.after.close.button.before.going.to.auto.fix.button.in.Run.local.spark.console.pptx)

**Comment by t-rufang** [comment]
Note: When the focus is at `close` button, we click twice, focus is not visible.
![image](https://user-images.githubusercontent.com/32627233/65660084-28f2e500-e060-11e9-8a28-3dc8264a9eab.png)

**Comment by t-rufang** [comment]
Fixed in #3669

**Comment by v-lasiva** [comment]
Regressed the issue in the updated version of IntelliJ having the following details,

Version: 2019.2.2

When the focus is on close button, For one tab the focus is still not visible.
As the issue is not resolved, Again re-activating the bug
Added attachment for reference.

[Bug_Focus order issue_Not fixed.pptx](https://github.com/microsoft/azure-tools-for-java/files/3845968/Bug_Focus.order.issue_Not.fixed.pptx)

**Comment by t-rufang** [comment]
If we open JAWS, we will find the invisible focus is at upper-right "close" button. If we press space at that time, the dialog will be closed.
Actually this behavior is introduced by `com.intellij.openapi.ui.DialogWrapper`. Please see another same issue in IntelliJ.
![issue_3606](https://user-images.githubusercontent.com/32627233/69129368-286b4d80-0ae9-11ea-98a3-73c6b2357fe6.gif)
As you can see when we tab once when focus at `close` button, the focus is invisible. And if we press space, the dialog will be closed.
Therefore, this bug should be owned by IntelliJ and it's out of the scope of our toolkit.

**Comment by t-rufang** [comment]
- Importance of experience: Minimal. When focus is on “Close” button, for 2 tabs focus is not visible before going to “Auto fix” button. This is a one-time thing for user. If they have fixed the dependency issue in this project before, they don't need to do them again.
- Quality of experience: Managable. When user hit an invisible focus, they just need to tab again, and twice later, they will hit a visible focus button.
- Alternate solutions: No alternate technologies or options can be leveraged.
- Evidence: All evidence is provided in this issue.
- [AAT Link](https://cela-ra-aat-prod.azurewebsites.net/#/workflow/56452990-0d38-45bc-b475-a62832ce0076)

**Comment by v-lasiva** [comment]
After discussing with SME, As they have approved closing this issue accordingly.

**Comment by vscjavaci** [comment]
Track internal issue by AB#1749708
